### DISCUSSION TITLE
Likert version of button-response or slider-response plugin

### DISCUSSION THREAD

**Discussion by amunn** [jspsych/jsPsych#888] [Q&A] [answered]
As I understand it, the survey-likert plugin assumes a set of questions and displays them all together, like a regular paper survey. I would like to do a likert response on a per trial basis, similar to the keyboard-response, button-response or slider response plugins, but such a beast doesn't seem to exist. Has anyone written one already?

If not, and I were to modify an existing plugin for that purpose, which should I use as a base?

Thanks

(I posted this in the Google group, but it seems you're moving things here. Apologies if this is out of place)

**Comment by jodeleeuw** [comment, marked as answer]
Hi @amunn ,

You can run a single likert question with the `survey-likert` plugin. It's generalized to support multiple questions, but supports single items.

```js
var trial = {
	type: 'survey-likert',
	questions: [
		{prompt: "I like vegetables.", labels: ["Strongly disagree", "Disagree", "Neutral", "Agree", "Strongly Agree"]}
	]
}
```

**Reply by amunn** [reply]
Thanks, but then I have no access to the typical parameters that I might want with the other plugins: stimulus duration, trial duration, prompt etc. These are sensible things for single trials and the likert plugin isn't designed with that in mind.

**Reply by jodeleeuw** [reply]
Could you use a slider response with only 5 or 7 levels?

**Reply by amunn** [reply]
Yes, that's what I've done. I hadn't realized that the slider levels combined with the step determine the degree to which the slider is analog. So this will work fine. Is there a way to simultaneously allow either keyboard or slider input for the same display (i.e., people would have the option of entering the numbers on the keyboard or using the slider?) Also is there documentation on how to style the slider itself apart from specifying its width?

**Reply by jodeleeuw** [reply]
> Is there a way to simultaneously allow either keyboard or slider input for the same display (i.e., people would have the option of entering the numbers on the keyboard or using the slider?)

This would require a plugin edit. You could add keyboard response functionality to the slider response plugin. I have [a recorded tutorial](https://www.youtube.com/watch?v=XQcsFwAmbiw) that covers creating a plugin with keyboard responses. It should cover what you need to know to add this functionality to an existing plugin.

> Also is there documentation on how to style the slider itself apart from specifying its width?

You can use any CSS styling you'd like here. We haven't parameterized these kinds of options in the plugin directly so we don't document them. But you can find tutorials on CSS modification of `<input type="range">` elements ([here's one](https://www.w3schools.com/howto/howto_js_rangeslider.asp); [here's another](http://danielstern.ca/range.css/#/)). You can add the relevant CSS directly to the `<head>` section via loading a stylesheet or adding `<style>` tags.

**Reply by amunn** [reply]
Thanks Josh, this is helpful. And jsPsych is a great set of tools, so thanks for creating it.
